ref/src/yppm_driver.f90

```diff
@@ -327,4 +344,79 @@ subroutine write_state(filename)
 
   end subroutine write_state
 
+  !------------------------------------------------------------------
+  ! interpolate_state
+  !
+  ! Increase the database by interpFactor.
+  !------------------------------------------------------------------
+  subroutine interpolate_state(interpFactor)
+
+    integer, intent(in) :: interpFactor
+
+    ! locals
+    ! Define new variables to replace the old variables.
+    ! Once the new variables are defined, they will replace the
+    ! old variables.
+
+    integer new_isd
+    integer new_ied
+    integer new_jsd
+    integer new_jed
+    integer new_js
+    integer new_je
+    real, allocatable :: new_cry(:, :)
+    real, allocatable :: new_q(:, :)
+    real, allocatable :: new_fy2(:, :)
+    real, allocatable :: new_dya(:, :)
+    integer :: odims(4), idims(4)
+
+    odims(1) = isd
+    odims(2) = ied
+    odims(3) = js
+    odims(4) = je + 1
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_cry(idims(1):idims(2), idims(3):idims(4)))
+    call interpolate_array(cry, odims, new_cry, idims, interpFactor)
+    
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_fy2(idims(1):idims(2), idims(3):idims(4)))
```

Comment:
This line is unnecessary since no allocation is happening.  The dimensions for the array below have already been computed by the call to `interpolate_allocate` above.